File: python/04-oop/oop.py
class Animal:
    
    num_legs = 4
    awake = True
    age = None

    def make_noise(self):
        print 'making noise'

    def sleep(self):
        print 'ZzZzZzZzzzzZZzz'
        self.awake = False

    def wakeup(self):
        print 'Waking up!!'
        self.awake = True

    def celebrate_birthday(self):
        self.age += 1
        print "Happy Birthday %s !!" % self.__class__.__name__ 


class Dog(Animal):

    def bark(self):
        print 'WuffWuff'

    def make_noise(self):
        self.bark()

class Cat(Animal):

    def miau(self):
        print 'Miau!!'

    def make_noise(self):
        self.miau()


    

File: ajax-php/00-work-in-progress/oop.js
// alert('ok');


Animal = function() {
	
	this.name = false;
	
	this.make_noise = function() {
		alert('making some noise');
	};

}

Dog = function() {
	
};

Dog.prototype = new Animal();

Dog.prototype.make_noise = function() {
	alert('WuffWuff');
};



//d = new Dog();
//d.make_noise();
